---
title: "Learning representations of chromatin contacts using a recurrent neural network identifies genomic drivers of conformation"
collection: publications
status: revision
permalink: /publications/hiclstm
excerpt: 'Despite the availability of chromatin conformation capture experiments, discerning the relationship between the 1D genome and 3D conformation remains a challenge, which limits our understanding of their affect on gene expression and disease. We propose Hi-C-LSTM, a method that produces low-dimensional latent representations that summarize intra-chromosomal Hi-C contacts via a recurrent long short-term memory neural network model. We find that these representations contain all the information needed to recreate the observed Hi-C matrix with high accuracy, outperforming existing methods. These representations enable the identification of a variety of conformation-defining genomic elements, including nuclear compartments and conformation-related transcription factors. They furthermore enable in-silico perturbation experiments that measure the influence of cis-regulatory elements on conformation.'
date: 2021-10-01
venue: 'Nature Communications'
paperurl: 'https://www.nature.com/articles/s41467-022-31337-w'
citation: 'Dsouza, K. B., Maslova, A., Al-Jibury, E., Merkenschlager, M., Bhargava, V. K., & Libbrecht, M. W. (2022). Learning representations of chromatin contacts using a recurrent neural network identifies genomic drivers of conformation. Nature Communications, 13(1), 1-19.'
---